mixer/adapter/kubernetesenv/kubernetesenv.go

```diff
@@ -138,30 +145,49 @@ func (b *builder) Build(ctx context.Context, env adapter.Env) (adapter.Handler,
 	if !exists {
 		path = paramsProto.KubeconfigPath
 	}
-	clientset, err := b.newClientFn(path, env)
+
+	config, err := b.newConfig(path, env)
```

Comment:
Do we really want to create the config here? If the path is the key, I think we can merge newConfig and newClientset in the same function (i.e. same one as the previous one).

Reply:
reverted to old behavior, with test added for cached controller.